purrr_basics/purrr_basics.Rmd

```diff
@@ -126,59 +129,366 @@ gapminder[1:6, "pop"]
+f <- mean
+```
+
+```{r slide-14-01}
+f
+```
+
+```{r slide-14-02}
+identical(x = mean, y = f)
+```
+
+
+---
+
+```{r slide-15-dplyr_across, out.height="100%", out.width="100%", echo = FALSE}
+knitr::include_graphics("img/dplyr_across.png")
+```
+
+.pull-right[Art by Allison Horst]
+
+---
+
+## `mutate(across())`
+
+--
+
+```{r slide-16-mutate-across-concept-map, eval=FALSE}
+mutate(
+  <DATA>, 
+  across(c(<VARIABLES>), list(<NAMES> = <FUNCTIONS>)) #<<
+)
+```
+
+---
+
+## `mutate(across())`
+
+```{r slide-17-mutate-across-example, eval=FALSE}
+mutate(
+  diamonds, 
+  across(c("carat", "depth"), list(sd = sd, mean = mean)) #<<
+)
+```
+
+---
+
+## `mutate(across())`
+
+```{r slide-18-1-mutate-across-example-code, eval=FALSE}
+mutate(
+  diamonds, 
+  across(c("carat", "depth"), list(sd = sd, mean = mean)) #<<
+)
+```
+
+.small[
+
+
+```{r slide-18-2-mutate-across-example-output, highlight.output = c(2,14), echo=FALSE}
+mutate(
+  diamonds, 
+  across(c("carat", "depth"), list(sd = sd, mean = mean))
+)
+```
+
+]
+
+---
+
+## `mutate(across(where()))`
+
+--
+
+```{r slide-19-mutate-across-where-example, eval=FALSE}
+mutate(
+  gapminder, 
+  across(where(is.numeric), list(mean = mean, median = median)) #<<
+)
+```
+
+
+---
+
+## `mutate(across(where()))`
+
+```{r slide-20-1-mutate-across-where-example-code, eval=FALSE}
+mutate(
+  gapminder, 
+  across(where(is.numeric), list(mean = mean, median = median)) #<<
+)
+```
+
+
+.small[
+
+```{r slide-20-2-mutate-across-where-example-output, highlight.output = c(2, 14, 15), echo=FALSE}
+mutate(
+  gapminder, 
+  across(where(is.numeric), list(mean = mean, median = median)) #<<
+)
+```
+
+]
+
+
+---
+
+## Review: `tidyselect`
+
+### 'Behind the scenes' workhorse for `dplyr::select()`,  `dplyr::pull()`, and `tidyr::pivot_` functions
+
+### `starts_with()`, `ends_with()`, `contains()`, `matches()`, etc.
+
+--
+
+.pull-left[
+
+.small[
+
+```{r slide-21-left-tidyselect}
+# column names contain a word
+select(diabetes, ends_with("ht"))
+```
+
+]
+
+]
+
+--
+
+.pull-right[
+
+.small[
+
+```{r slide-21-right-tidyselect}
+# regular expression
+select(diabetes, matches("\\d")) 
+```
+
+]
+
+]
+
+---
+
+## `mutate(across())` & `summarise()`
+
+### Perform group-wise summaries with `mutate(across())`, `group_by()` and `summarise()`
+
+```{r slide-22-mutate-across-summarise-concept-map, eval=FALSE}
+<DATA> %>% 
+  group_by(<GROUPING VAR>) #<<
+  summarise(across(c(<VARIABLES>), list(<NAMES> = <FUNCTIONS>))) 
+```
+
+---
+
+## `mutate(across())` & `summarise()`
+
+### Combine with `tidyselect()` helpers
+
+```{r slide-23-mutate-across-tidyselect-summarise-concept-map, eval=FALSE}
+<DATA> %>% 
+  group_by(<GROUPING VAR>) 
+  summarise(across(contains(<VARIABLES>), list(<NAMES> = #<<
+      <FUNCTIONS>))) 
+```
+
+---
+
+## `mutate(across())` & `summarise()`
+
+```{r slide-24-code, eval=FALSE}
+gapminder %>% 
+  group_by(continent) %>% 
+    summarise(across(c("lifeExp", "gdpPercap"), list(med = median, #<<
```

Comment:
let's vary this one a bit, too, and do `summarise(across(c("lifeExp", "gdpPercap"), median)` to show we don't always need a list